### ISSUE TITLE
`team.invest()` doesn't set action.llm.cost_manager

### ISSUE THREAD

**Issue by azurewtl** [geekan/MetaGPT#1086]
**Bug description**
``` python
team = Team(context=ctx)
team.invest(investment=args.cost)
team.hire(
    [
        Manager(),
        Searcher(),
        Researcher(),
        Editor(),
        Reviewer(),
        Human(is_human=True),
    ]
)
```
`team.invest()` doesn't set action.llm.cost_manager, 

**Bug solved method**
I think ContextMixin should be implemented as a singleton?
When instanciate new Action, it subclass `ContextMixin`. And I think the problem might be in this property. `line = 78-83`

``` python
@property
def context(self) -> Context:
    """Role context: role context > context"""
    if self.private_context:
        return self.private_context
    return Context()
```
It returns a new context regardless of what `cost_manager` you set from `team.invest()`, because when we get llm from action instance, it use `self.context`
line 94-95
``` python
if not self.private_llm:
    self.private_llm = self.context.llm_with_cost_manager_from_llm_config(self.config.llm)
```

**Environment information**

- LLM type and model name:
- System version: metagpt 0.7.6
- Python version: python 3.10

**Comment by geekan** [comment]
This was originally designed for priority of Context: Global > Env > Role > Action
But I noticed that the global Context was removed in a certain version, so there were corresponding modifications in a Commit.
I would like to raise a discussion, is it more reasonable to provide a GlobalEnv and use its Context by default?

![image](https://github.com/geekan/MetaGPT/assets/2707039/7a510983-82a1-4975-a7b3-2a55cfd270ec)

**Comment by iorisa** [comment]
fixed by:  #1031
